com.unity.render-pipelines.high-definition/Runtime/Material/MaterialExtension.cs

```diff
@@ -190,18 +199,296 @@ public static void UpdateEmissiveColorFromIntensityAndEmissiveColorLDR(this Mate
+            SetDiffusionProfileAsset(material, profile, HDShaderIDs._DiffusionProfileAsset);
+#endif
+        }
+
+        /// <summary>Set a Diffusion profile on a Shader Graph material.</summary>
+        /// <param name="material">The material to change.</param>
+        /// <param name="profile">The Diffusion Profile Asset.</param>
+        /// <param name="referenceName">The reference name of the Diffusion Profile property in the Shader Graph.</param>
+        public static void SetDiffusionProfileShaderGraph(Material material, DiffusionProfileSettings profile, string referenceName)
+        {
+            float hash = profile != null ? HDShadowUtils.Asfloat(profile.profile.hash) : 0;
+            material.SetFloat(referenceName, hash);
+
+#if UNITY_EDITOR
+            SetDiffusionProfileAsset(material, profile, Shader.PropertyToID(referenceName + "_Asset"));
+#endif
+        }
+
+#if UNITY_EDITOR
+        internal static void SetDiffusionProfileAsset(Material material, DiffusionProfileSettings profile, int assetPropertyId, int index = 0)
+        {
+            Vector4 guid = Vector3.zero;
+            if (profile != null)
+                guid = HDUtils.ConvertGUIDToVector4(UnityEditor.AssetDatabase.AssetPathToGUID(UnityEditor.AssetDatabase.GetAssetPath(profile)));
+            material.SetVector(assetPropertyId, guid);
+
+            var externalRefs = MaterialExternalReferences.GetMaterialExternalReferences(material);
+            externalRefs.SetDiffusionProfileReference(index, profile);
+        }
+
+        /// <summary>Get the Diffusion profile on Lit shaders.</summary>
+        /// <param name="material">The material to access.</param>
+        /// <returns>The Diffusion Profile Asset.</returns>
+        public static DiffusionProfileSettings GetDiffusionProfile(Material material)
+        {
+            return GetDiffusionProfileAsset(material, HDShaderIDs._DiffusionProfileAsset);
+        }
+
+        /// <summary>Get the Diffusion profile on a Shader Graph material.</summary>
+        /// <param name="material">The material to access.</param>
+        /// <returns>The Diffusion Profile Asset.</returns>
+        public static DiffusionProfileSettings GetDiffusionProfileShaderGraph(Material material)
+        {
+            return GetDiffusionProfileAsset(material, HDShaderIDs._DiffusionProfileAsset);
+        }
+
+        internal static DiffusionProfileSettings GetDiffusionProfileAsset(Material material, int assetPropertyId)
+        {
+            string guid = HDUtils.ConvertVector4ToGUID(material.GetVector(assetPropertyId));
+            return UnityEditor.AssetDatabase.LoadAssetAtPath<DiffusionProfileSettings>(UnityEditor.AssetDatabase.GUIDToAssetPath(guid));
+        }
+#endif
+
+        // this will work on ALL shadergraph-built shaders, in memory or asset based
+        internal static bool IsShaderGraph(Material material)
+        {
+            var shaderGraphTag = material.GetTag("ShaderGraphShader", false, null);
```

Comment:
I never noticed that shader graph was adding a tag ShaderGraphShader, is it new?

Reply:
put a reference that you duplicate the function